packages/react/menu/src/Menu.tsx

```diff
@@ -692,19 +731,32 @@ const MenuSubTrigger = React.forwardRef((props, forwardedRef) => {
           const contentRect = context.content?.getBoundingClientRect();
           if (contentRect) {
             // TODO: make sure to update this when we change positioning logic
-            const side = context.content?.dataset.side;
-            const contentEdge = side === 'right' ? contentRect.left : contentRect.right;
-            contentContext.onPointerGraceAreaChange([
-              { x: event.clientX, y: event.clientY },
-              { x: contentEdge, y: contentRect.top },
-              { x: contentEdge, y: contentRect.bottom },
-            ]);
-
-            pointerGraceDurationTimer.current = window.setTimeout(
+            const side = context.content?.dataset.side as Side;
+            const rightSide = side === 'right';
+            const bleed = rightSide ? -5 : +5;
```

Comment:
This bleed value is arbitrary but ensures that the exit point is consistently inside the polygon across browsers and rtl.